src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/correlation/CorrelationExtractor.java

```diff
@@ -0,0 +1,327 @@
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.transform.TransformerException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.jmeter.extractor.BoundaryExtractor;
+import org.apache.jmeter.extractor.HtmlExtractor;
+import org.apache.jmeter.extractor.RegexExtractor;
+import org.apache.jmeter.extractor.XPath2Extractor;
+import org.apache.jmeter.extractor.json.jsonpath.JSONPostProcessor;
+import org.apache.jmeter.gui.GuiPackage;
+import org.apache.jmeter.protocol.http.correlation.extractordata.BoundaryExtractorData;
+import org.apache.jmeter.protocol.http.correlation.extractordata.ExtractorData;
+import org.apache.jmeter.protocol.http.correlation.extractordata.HtmlExtractorData;
+import org.apache.jmeter.protocol.http.correlation.extractordata.JsonPathExtractorData;
+import org.apache.jmeter.protocol.http.correlation.extractordata.RegexExtractorData;
+import org.apache.jmeter.protocol.http.correlation.extractordata.XPath2ExtractorData;
+import org.apache.jmeter.samplers.SampleResult;
+import org.apache.jmeter.testelement.TestElement;
+import org.apache.jmeter.util.JMeterUtils;
+import org.apache.jmeter.visualizers.CorrelationRecorder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CorrelationExtractor {
+
+    private static final Logger log = LoggerFactory.getLogger(CorrelationExtractor.class);
+
+    private static final String APPLICATION_JSON = "application/json"; //$NON-NLS-1$
+    private static final String APPLICATION_XML = "application/xml"; //$NON-NLS-1$
+    private static final String TEXT_HTML = "text/html"; //$NON-NLS-1$
+    private static final String TEXT_XML = "text/xml"; //$NON-NLS-1$
+
+    private static final String OTHER = "other"; //$NON-NLS-1$
+    private static final String HEADER = "header"; //$NON-NLS-1$
+    private static final String BOUNDARY = "boundary"; //$NON-NLS-1$
+
+    private static List<ExtractorData> listOfExtractors = new ArrayList<>();
+
+    private CorrelationExtractor() {}
+
+    /**
+     * Read the response and create the extractor tags.
+     *
+     * @param parameters   List of correlation candidate parameters
+     * @param parameterMap Map containing correlation candidates and their values
+     */
+    public static void readResponse(List<String> parameters, Map<String, String> parameterMap) {
+        log.debug("Start finding parameters in response data and create extractors.");
+        // only one extractor for one parameter to be created
+        for (String parameter : parameters) {
+            ExtractorData extractor = createExtractorForParameter(parameter, parameterMap.get(parameter));
+            if (extractor != null) {
+                getListOfExtractor().add(extractor);
+            }
+        }
+        // update the JMX file with extractor tags
+        log.debug("Processing of sample results ended.");
```

Comment:
can you add some context to this debug message to make it useful ?

Reply:
Described the context properly.

Reply:
We have added context information in the comment in order to make it useful.